pkg/gohai/memory/memory.go

```diff
@@ -6,48 +6,27 @@
 // Package memory regroups collecting information about the memory
 package memory
 
-// Memory holds memory metadata about the host
-type Memory struct {
-	// TotalBytes is the total memory for the host in byte
-	TotalBytes uint64
-	// SwapTotalBytes is the swap memory size in byte (Unix only)
-	SwapTotalBytes uint64
-}
-
-const name = "memory"
+import (
+	"github.com/DataDog/datadog-agent/pkg/gohai/utils"
+)
 
-// Name returns the name of the package
-func (memory *Memory) Name() string {
-	return name
+// Info holds memory metadata about the host
+type Info struct {
+	// TotalBytes is the total memory for the host in byte
+	TotalBytes utils.Value[uint64] `json:"total"`
+	// SwapTotalBytes is the swap memory size in kilobyte (Unix only)
+	SwapTotalKb utils.Value[uint64] `json:"swap_total" unit:"kb"`
```

Comment:
`SwapTotalKb` is now always displayed as an integer with `kb` unit. The format was different on Darwin (in particular it was locale-dependent), which caused issues.